openedx/core/djangoapps/user_authn/views/tests/test_login.py

```diff
@@ -573,6 +579,63 @@ def _assert_not_in_audit_log(self, mock_audit_log, level, log_strings):
         for log_string in log_strings:
             self.assertNotIn(log_string, format_string)
 
+    @ddt.data(
+        {
+            'switch_enabled': False,
+            'whitelisted': False,
+            'allowed_domain': 'edx.org',
+            'success': True
+        },
+        {
+            'switch_enabled': False,
+            'whitelisted': True,
+            'allowed_domain': 'edx.org',
+            'success': True
+        },
+        {
+            'switch_enabled': True,
+            'whitelisted': False,
+            'allowed_domain': 'edx.org',
+            'success': False
+        },
+        {
+            'switch_enabled': True,
+            'whitelisted': False,
+            'allowed_domain': 'fake.org',
+            'success': True
+        },
+        {
+            'switch_enabled': True,
+            'whitelisted': True,
+            'allowed_domain': 'edx.org',
+            'success': True
+        },
+    )
+    @ddt.unpack
+    def test_login_for_user_auth_flow(self, switch_enabled, whitelisted, allowed_domain, success):
+        """
+        Verify that `login._check_user_auth_flow` works as expected.
+        """
+        site = self.set_up_site('edx.org', {
+            'SITE_NAME': 'edx.org',
+            'THIRD_PARTY_AUTH_ONLY_DOMAIN': allowed_domain,
+            'THIRD_PARTY_AUTH_ONLY_PROVIDER': 'Google'
+        })
+
+        if whitelisted:
+            AllowedAuthUser.objects.create(site=site, email=self.user_email)
+        else:
+            AllowedAuthUser.objects.filter(site=site, email=self.user_email).delete()
+
+        with authn_waffle().override(ENABLE_LOGIN_USING_THIRDPARTY_AUTH_ONLY, switch_enabled):
+            value = None if success else 'As an edx.org user, You must login with your edx.org Google account.'
```

Comment:
This expected message should account for the `allowed_domain` in the test data (and you should probably test with another domain).